### PR TITLE
Cal/telemetry

### PR DESCRIPTION
**Pull request by McCallisterRomer**
-Added function profiling
-Removed some methods from Editor and moved most imgui code to another file

### CODE REVIEW

File: nc/source/debug/Profile.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    void Push(FuncId id)
+    {
+        internal::openProfiles.push_back(id);
+        try
+        {
+            auto& data = internal::data.at(id);
+            ++data.callCount;
+            data.timer.Start();
+        }
+        catch(...)
```
Comment (McCallisterRomer) [inline nc/source/debug/Profile.cpp]: Can catch std::exception here instead of ...

File: nc/source/ui/editor/EditorControls.h [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            ImGui::Text("Scene Graph");
+            ImGui::Spacing();
+            filter.Draw("##filterlabel", 2.0f * DefaultItemWidth);
+            ImGui::Separator();
+            ImGui::Separator();
+            ImGui::Spacing();
+
+            if(ImGui::BeginChild("EntityList"))
+            {
+                auto uidScope = 0u;
```
Comment (McCallisterRomer) [inline nc/source/ui/editor/EditorControls.h]: Should be possible to use handle as the id here.

**Comment by jaremieromer** [comment]
Can we change Profile.h to Profiler.h? Even knowing what this pull request was for I was confused by the contents of Profile (maybe because the filter values are User and Engine) and didn't click that it was a profiler until I looked at how it was being used.

File: nc/source/component/Transform.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ namespace nc
             Quaternion GetRotation() const;
             Vector3 GetScale() const;
 
-            DirectX::XMMATRIX GetTransformationMatrix() const;
+            DirectX::FXMMATRIX GetTransformationMatrix() const;
```
Comment (jaremieromer) [inline nc/source/component/Transform.h:28]: Why FXMMATRIX?

File: nc/source/ui/editor/Editor.cpp [hunk 4]
```diff
@@ -1,223 +1,67 @@
...
-    const auto MainWindowFlags = ImGuiWindowFlags_NoBackground |
+    constexpr auto MainWindowFlags = ImGuiWindowFlags_NoBackground |
                                  ImGuiWindowFlags_MenuBar |
                                  ImGuiWindowFlags_NoResize |
                                  ImGuiWindowFlags_NoScrollbar |
                                  ImGuiWindowFlags_NoMove;
-
-    void DrawComponentControl(nc::ComponentBase* comp)
+    
+    namespace hotkey
```
Comment (jaremieromer) [inline nc/source/ui/editor/Editor.cpp:16]: This is nice

File: nc/source/ui/editor/EditorControls.h [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+#ifdef NC_EDITOR_ENABLED
+#include "Profile.h"
+#include "Ecs.h"
+#include "graphics/d3dresource/GraphicsResourceManager.h"
+#include "imgui/imgui.h"
+
+
+namespace nc::ui::editor::controls
+{
+    const auto TitleBarHeight = 40.0f;
```
Comment (jaremieromer) [inline nc/source/ui/editor/EditorControls.h:11]: Not sure if worth it for IMGUI, but we should eventually not hardcode these sizes.

**Review by jaremieromer — APPROVED** [review]

File: nc/source/component/Transform.h (hunk 3 above)
Comment (McCallisterRomer) [reply nc/source/component/Transform.h:28]: It is a typedef for const & or const depending on platform.

File: nc/source/ui/editor/EditorControls.h (hunk 5 above)
Comment (McCallisterRomer) [reply nc/source/ui/editor/EditorControls.h:11]: Agreed. Once we've settled on a permanent solution, we'll figure out a proper way to do layout and scaling.

**Pull request merged**